[2024-01-31T10:14:15.294+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: complex_music.convert_to_csv manual__2024-01-31T10:14:08.193621+00:00 [queued]>
[2024-01-31T10:14:15.298+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: complex_music.convert_to_csv manual__2024-01-31T10:14:08.193621+00:00 [queued]>
[2024-01-31T10:14:15.298+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 2
[2024-01-31T10:14:15.303+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): convert_to_csv> on 2024-01-31 10:14:08.193621+00:00
[2024-01-31T10:14:15.309+0000] {standard_task_runner.py:60} INFO - Started process 222 to run task
[2024-01-31T10:14:15.311+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'complex_music', 'convert_to_csv', 'manual__2024-01-31T10:14:08.193621+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/complex_music.py', '--cfg-path', '/tmp/tmpajnk9uwj']
[2024-01-31T10:14:15.312+0000] {standard_task_runner.py:88} INFO - Job 6: Subtask convert_to_csv
[2024-01-31T10:14:15.335+0000] {task_command.py:423} INFO - Running <TaskInstance: complex_music.convert_to_csv manual__2024-01-31T10:14:08.193621+00:00 [running]> on host 7c676a75bc22
[2024-01-31T10:14:15.352+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-01-31T10:14:15.370+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='complex_music' AIRFLOW_CTX_TASK_ID='convert_to_csv' AIRFLOW_CTX_EXECUTION_DATE='2024-01-31T10:14:08.193621+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-01-31T10:14:08.193621+00:00'
[2024-01-31T10:14:15.373+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/compat/_optional.py", line 142, in import_optional_dependency
    module = importlib.import_module(name)
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'openpyxl'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/complex_music.py", line 44, in _convert_to_csv
    data_xls = pd.read_excel(src, 'dft', dtype=str, index_col=None)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_base.py", line 478, in read_excel
    io = ExcelFile(io, storage_options=storage_options, engine=engine)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_base.py", line 1513, in __init__
    self._reader = self._engines[engine](self._io, storage_options=storage_options)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_openpyxl.py", line 548, in __init__
    import_optional_dependency("openpyxl")
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/compat/_optional.py", line 145, in import_optional_dependency
    raise ImportError(msg)
ImportError: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
[2024-01-31T10:14:15.379+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=complex_music, task_id=convert_to_csv, execution_date=20240131T101408, start_date=20240131T101415, end_date=20240131T101415
[2024-01-31T10:14:15.386+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 6 for task convert_to_csv (Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.; 222)
[2024-01-31T10:14:15.404+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-01-31T10:14:15.417+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
